src/shell-integration/nushell/vendor/autoload/ghostty-integration.nu

```diff
@@ -0,0 +1,103 @@
+# Enables SSH environment variable compatibility.
+# Converts TERM from xterm-ghostty to xterm-256color
+# and propagates COLORTERM, TERM_PROGRAM, and TERM_PROGRAM_VERSION
+# check your sshd_config on remote host to see if these variables are accepted
+def set_ssh_env []: nothing -> record<ssh_term: string, ssh_opts: list<string>> {
+    return {ssh_term: "xterm-256color", ssh_opts: ["-o", "SetEnv COLORTERM=truecolor", "-o", "SendEnv TERM_PROGRAM TERM_PROGRAM_VERSION"]}
+}
+
+# Enables automatic terminfo installation on remote hosts.
+# Attempts to install Ghostty's terminfo entry using infocmp and tic when
+# connecting to hosts that lack it. 
+# Requires infocmp to be available locally and tic to be available on remote hosts.
+# Caches installations to avoid repeat installations.
+def set_ssh_terminfo [ssh_opts: list<string>, ssh_args: list<string>] {
```

Comment:
For consistency, I think we should also specify the `record<...>` return type here. If it feels too verbose, we could also just say `record`.

There unfortunately doesn't appear to be a way to define reusable named types in Nu.